### PR TITLE
Enhance handling of polymorphic foreign key-like columns in `UnindexedForeignKeys`

### PR DESCRIPTION
**Pull request by fatkodima**
Accept polymorphic foreign key-like columns which are a prefix of a larger index. Previously, only exact match on index columns was made.

### CODE REVIEW

File: test/active_record_doctor/detectors/unindexed_foreign_keys_test.rb [hunk 1]
```diff
@@ -14,6 +14,34 @@ def test_unindexed_foreign_key_is_reported
     OUTPUT
   end
 
+  def test_unindexed_polymorphic_foreign_key_is_reported
+    skip("MySQL always indexes foreign keys") if mysql?
```
Comment (gregnavis) [inline test/active_record_doctor/detectors/unindexed_foreign_keys_test.rb:18]: It seems this restriction can be removed, as we're not dealing with real foreign keys.

File: test/active_record_doctor/detectors/unindexed_foreign_keys_test.rb [hunk 2]
```diff
@@ -14,6 +14,34 @@ def test_unindexed_foreign_key_is_reported
...
+
+    create_table(:notes) do |t|
+      t.integer :notable_id
+      t.string :notable_type
+    end
+
+    assert_problems(<<~OUTPUT)
+      add an index on notes.notable_id - foreign keys are often used in database lookups and should be indexed for performance reasons
```
Comment (gregnavis) [inline test/active_record_doctor/detectors/unindexed_foreign_keys_test.rb:26]: This error message is incorrect and should mention `notable_type` too:

```
add an index on notes(notable_type, notable_id) - foreign keys are often used in database lookups and should be indexed for performance reasons
```

**Comment by fatkodima** [comment]
Made suggested changes. The PR became a little bigger, then was initially.

**Comment by fatkodima** [comment]
Rebased.

Yeah, only a few more left 😄 🎉

**Review by gregnavis — APPROVED** [review]

**Pull request merged**